crates/sui-indexer-alt-jsonrpc/src/api/objects/filter.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Context as _;
+use diesel::{
+    dsl::sql,
+    sql_types::{BigInt, Bool, Bytea},
+    BoolExpressionMethods, ExpressionMethods, JoinOnDsl, QueryDsl,
+};
+use move_core_types::language_storage::StructTag;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use sui_indexer_alt_schema::{objects::StoredOwnerKind, schema::obj_info};
+use sui_json_rpc_types::{Page as PageResponse, SuiObjectDataOptions};
+use sui_types::{
+    base_types::{ObjectID, SuiAddress},
+    Identifier,
+};
+
+use crate::{
+    error::{rpc_bail, RpcError},
+    paginate::{BcsCursor, Cursor as _, Page},
+    Context,
+};
+
+use super::{error::Error, ObjectsConfig};
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema, Default)]
+#[serde(rename_all = "camelCase", rename = "ObjectResponseQuery", default)]
+pub(crate) struct SuiObjectResponseQuery {
+    /// If None, no filter will be applied
+    pub filter: Option<SuiObjectDataFilter>,
+    /// config which fields to include in the response, by default only digest is included
+    pub options: Option<SuiObjectDataOptions>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
+pub(crate) enum SuiObjectDataFilter {
```

Comment:
I'm really glad that we decided not to use the existing APIs defined in `sui-json-rpc-api`. Otherwise we'd have to deal with all the filters (MatchAll, MatchAny..) we don't support.

Reply:
Yeah, I'm actually going to go back and swap out the type for query transaction blocks as well, for the same reason.